src/sys/inotify.rs

```diff
@@ -0,0 +1,225 @@
+//! Monitoring API for filesystem events.
+//!
+//! Documentation
+//! =============
+//!
+//! Inotify is a Linux-only API to monitor filesystems events.
+//!
+//! For more documentation, please read [inotify(7)](http://man7.org/linux/man-pages/man7/inotify.7.html).
+//! 
+//! Example
+//! =======
+//!
+//! * Monitor all events happening in directory "test":
+//! ```no_run
+//! # use nix::sys::inotify::{EventFlags,InotifyInitFlags,Inotify};
+//! #
+//! // We create a new inotify instance.
+//! let instance = Inotify::init(InotifyInitFlags::empty()).unwrap();
+//!
+//! // We add a new watch on directory "test" for all events.
+//! let wd = instance.add_watch("test", EventFlags::IN_ALL_EVENTS).unwrap();
+//!
+//! loop {
+//!     // We read from our inotify instance for events.
+//!     let events = instance.read_events().unwrap();
+//!     println!("Events: {:?}", events);
+//! }
+//! ```
+
+use libc;
+use libc::{
+    c_char,
+    c_int,
+    uint32_t
+};
+use std::ffi::{OsString,OsStr,CStr};
+use std::os::unix::ffi::OsStrExt;
+use std::mem::size_of;
+use std::os::unix::io::{RawFd,AsRawFd,FromRawFd};
+use unistd::read;
+use Result;
+use NixPath;
+use errno::Errno;
+
+libc_bitflags! {
+    /// Configuration options for [`inotify_add_watch`](fn.inotify_add_watch.html).
+    pub struct EventFlags: uint32_t {
```

Comment:
According to our guidelines in CONVENTIONS.md, this struct should be named `InFlags`.  However, the guidelines say that the `InotifyInitFlags` should be named the same way.  How about renaming them `AddWatchFlags` and `InitFlags`?  That way it's clear which functions they should be used with, and they don't repeat the "inotify" part of the module name.

Reply:
I like the name EventFlags as the flags are used not only in `add_watch` but also as a result in the struct `InotifyEvent`.